python/Gaffer/PythonExpressionEngine.py

```diff
@@ -100,7 +103,13 @@ def apply( self, proxyOutput, topLevelProxyOutput, value ) :
 
 		value = _extractPlugValue( proxyOutput, topLevelProxyOutput, value )
 		if value is None :
-			raise TypeError( "Unsupported value type \"%s\"" % type( value ).__name__ )
+			raise TypeError( "Error on node \"{0}\", Unsupported value type \"{1}\" while connection input plug \"{2}\" and output plug \"{3}\"".format(
```

Comment:
"while connection input plug X and output plug Y" doesn't seem to describe the situation here. Both plugs are outputs, and in any case they're not being connected - we're trying to set the value. Maybe just "Unsupported value type V for plug X"?